DxDispatch/src/model/ImageReaderWriter.cpp

```diff
@@ -0,0 +1,271 @@
+            bitmapSource.Get(),
+            dstTensorInfo.width,
+            dstTensorInfo.height,
+            WICBitmapInterpolationModeHighQualityCubic
+        ));
+
+        bitmapSource = scaler;
+    }
+
+    // WIC CopyPixels writes to an interleaved (HWC) buffer with 8 bits (uint8) per channel element.
+    std::vector<std::byte> pixelBuffer(dstTensorInfo.height * dstTensorInfo.width * dstTensorInfo.channels);
+    WICRect pixelBufferRect = { 0, 0, static_cast<INT>(dstTensorInfo.width), static_cast<INT>(dstTensorInfo.height) };
+    const uint32_t pixelBufferStride = dstTensorInfo.width * dstTensorInfo.channels * sizeof(uint8_t);
+
+    THROW_IF_FAILED(bitmapSource->CopyPixels(
+        &pixelBufferRect,
+        pixelBufferStride,
+        static_cast<uint32_t>(pixelBuffer.size()),
+        reinterpret_cast<BYTE*>(pixelBuffer.data())
+    ));
+
+    switch (dstTensorInfo.dataType)
+    {
+        case DML_TENSOR_DATA_TYPE_FLOAT32: return ConvertPixelsToTensor<float>(pixelBuffer, dstTensorInfo);
+        case DML_TENSOR_DATA_TYPE_FLOAT16: return ConvertPixelsToTensor<half_float::half>(pixelBuffer, dstTensorInfo);
+
+        default: throw std::invalid_argument("Unsupported data type");
+    }
+}
+
+// Converts a tensor to a pixel buffer (HWC layout with uint8 components).
+template <typename T>
+std::vector<std::byte> ConvertTensorToPixels(std::span<const std::byte> srcRaw, const ImageTensorInfo& tensorInfo)
+{
+    std::span<const T> src(reinterpret_cast<const T*>(srcRaw.data()), srcRaw.size_bytes() / sizeof(T));
+
+    std::vector<std::byte> dst(tensorInfo.height * tensorInfo.width * tensorInfo.channels);
+
+    for (size_t pixelIndex = 0; pixelIndex < tensorInfo.height * tensorInfo.width; pixelIndex++)
+    {
+        for (size_t channelIndex = 0; channelIndex < tensorInfo.channels; channelIndex++)
+        {
+            // dst is always HWC layout.
+            size_t dstOffset = pixelIndex * tensorInfo.channels + channelIndex;
+
+            // src is either HWC or CHW layout.
+            size_t srcOffset = tensorInfo.layout == ImageTensorLayout::NHWC ?
+                dstOffset :
+                pixelIndex + channelIndex * tensorInfo.height * tensorInfo.width;
+
+            dst[dstOffset] = static_cast<std::byte>(std::max(0.0f, std::min(1.0f, (float)src[srcOffset])) * 255.0f);
+        }
+    }
+
+    return dst;
+}
+
+void WriteTensorToImage(std::filesystem::path dstPath, std::span<const std::byte> srcData, const ImageTensorInfo& srcTensorInfo)
+{
+    WICPixelFormatGUID desiredImagePixelFormat = GUID_WICPixelFormatDontCare;
+
+    switch (srcTensorInfo.channelOrder)
+    {
+        case ImageTensorChannelOrder::RGB: desiredImagePixelFormat = GUID_WICPixelFormat24bppRGB; break;
+        case ImageTensorChannelOrder::RGBA: desiredImagePixelFormat = GUID_WICPixelFormat32bppRGBA; break;
+        case ImageTensorChannelOrder::BGR: desiredImagePixelFormat = GUID_WICPixelFormat24bppBGR; break;
```

Comment:
```suggestion
        case ImageTensorChannelOrder::BGR: desiredImagePixelFormat = GUID_WICPixelFormat24bppBGR; break;
        case ImageTensorChannelOrder::BGRA: desiredImagePixelFormat = GUID_WICPixelFormat24bppBGRA; break;
```

`GUID_WICPixelFormat32bppBGRA` for completeness? Despite the prevalence of RGBA in nomenclature (like CSS and HTML hex codes), BGRA has always been the default for DIB's when created, and it's the default surface format for video frame buffers on Windows.